### PR TITLE
Cleanup for Kubebuilder Beta.1

### PR DESCRIPTION
**Pull request by shaneutt**
This is a small cleanup PR to update KIC 2.0 components for [this morning's release][0] of [Kubebuilder][1], which included some backwards incompatible changes.

[0]:https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.0.0-beta.1
[1]:https://github.com/shaneutt/kubernetes-ingress-controller/pull/2

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ go 1.16
 // -----------------------------------------------------------------------------
 
 // client-go does not autoresolve versions correctly due to a historical versioning problem upstream
-replace k8s.io/client-go => k8s.io/client-go v0.20.3
+replace k8s.io/client-go => k8s.io/client-go v0.20.5
```
Comment (mflendrich) [inline go.mod:13]: Just realized that, when reviewing, I missed that 50a5bb9 re-added the `replace` directive that af95eab removed right before :facepalm: .

With Go 1.16, we should be able to remove this `replace` once and for all. This requires bumping Go to 1.16 in KTF and removing the replace directive there. Let's talk about this in the upcoming zoom.

Comment (shaneutt) [reply go.mod:13]:
```suggestion
```

File: go.mod [hunk 2]
```diff
@@ -10,7 +10,6 @@ go 1.16
 // -----------------------------------------------------------------------------
 
 // client-go does not autoresolve versions correctly due to a historical versioning problem upstream
```
Comment (shaneutt) [inline go.mod:12]:
```suggestion
```

File: go.mod [hunk 3]
```diff
@@ -9,8 +9,6 @@ go 1.16
 // associated with it that explains why the replacement is necessary.
 // -----------------------------------------------------------------------------
```
Comment (shaneutt) [inline go.mod:11]:
```suggestion
```

**Review by mflendrich — APPROVED** [review]

**Pull request merged**